src/main/scala/com/example/Registry/Registry.scala

```diff
@@ -0,0 +1,136 @@
+package com.example.Registry
+
+import java.sql.Timestamp
+
+import akka.actor.typed.scaladsl.Behaviors
+import akka.actor.typed.{ActorRef, Behavior}
+import akka.pattern.StatusReply
+import com.example.Service.Reservation
+import pureconfig.{ConfigReader, ConfigSource}
+
+import scala.concurrent.Await
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.duration.{Duration, SECONDS}
+import scala.util.{Failure, Success}
+
+import pureconfig.generic.auto._
+
+final case class ScreeningTitleDB(movieTitle: String, screeningId: Int, screeningTime: Timestamp)
+final case class ScreeningTitlesDB(screeningTitles: Seq[ScreeningTitleDB])
+
+final case class ScreeningScreeningRoomDB(screeningId: Int, screeningRoomId: Int, rows: Int, seatsPerRow: Int)
+
+final case class ReservationsScreeningDB(reservations: Seq[(Int, Int)])
+
+final case class ScreeningTime(time: Timestamp)
+
+final case class ReservationCreatedDB(insertedRows: Int)
+
+object Registry {
+  sealed trait RegistryCommand
+  final case class GetScreeningTitlesDB(from: Timestamp, until: Timestamp, replyTo: ActorRef[StatusReply[ScreeningTitlesDB]]) extends RegistryCommand
+
+  final case class GetScreeningScreeningRoomDB(id: Int, replyTo: ActorRef[StatusReply[Option[ScreeningScreeningRoomDB]]]) extends RegistryCommand
+
+  final case class CreateReservationDB(reservation: Reservation, replyTo: ActorRef[StatusReply[ReservationCreatedDB]]) extends RegistryCommand
+
+  final case class GetScreeningTimeDB(id: Int, replyTo: ActorRef[StatusReply[Option[ScreeningTime]]]) extends RegistryCommand
+
+  final case class GetReservationsScreeningDB(screeningId: Int, replyTo: ActorRef[StatusReply[ReservationsScreeningDB]]) extends RegistryCommand
+
+  import slick.jdbc.H2Profile.api._
+  val db = Database.forConfig("h2mem1")
+
+  val movies = MovieSchema.movies
+  val screeningRooms = ScreeningRoomSchema.screeningRooms
+  val screenings = ScreeningSchema.screenings
+  val reservations = ReservationSchema.reservations
+
+  def apply(): Behavior[RegistryCommand] = {
+
+    implicit val timestampReader = ConfigReader[String].map(s => Timestamp.valueOf(s))
+
+    Await.result(
+      db.run {(
+          movies.schema ++
```

Comment:
what will happen here after running the app with database with already applied schema?

Reply:
it would throw an sql exception, instead of .create method, it should be .createIfNotExists or dropIfExists before

Reply:
yea, i should use a migration tool